checkov/arm/utils.py

```diff
@@ -60,3 +60,32 @@ def get_files_definitions(
             parsing_errors.append(os.path.normpath(file))
 
     return definitions, definitions_raw, parsing_errors
+
+
+def extract_resource_name_from_resource_id_func(resource_id: str) -> str:
+    '''
+        Examples:
+            resourceId('Microsoft.Network/virtualNetworks/', virtualNetworkName) -> virtualNetworkName
+    '''
+    return clean_string(resource_id.split(',')[-1].split(')')[0])
+
+
+def extract_resource_name_from_reference_func(reference: str) -> str:
```

Comment:
can you create unit tests for this func? seems like you already have multiple examples you can check

Reply:
Done